.infrastructure/helm-chart/todoapp/templates/app/secret.yml

```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .Chart.Name }}-secret
+  namespace: {{ .Values.common.namespace }}
+type: Opaque
+data:
+  {{- range $k, $v := .Values.secret }}
+  {{ $k }}: {{ $v | b64enc | quote }}
+  {{- end }}
```

Comment:
Ensure that `.Values.secret` is defined in your `values.yaml` file. This is necessary for populating the Secret with the correct data.